### PR TITLE
[INLONG-12148][Agent] Replace /bin/sh -c with ProcessBuilder

### LINKED ISSUE #12148: [Improve] [Agent] Replace /bin/sh -c with ProcessBuilder
### Background
`ExcuteLinux.exeCmd()` currently executes commands via `/bin/sh -c` with no input filtering, allowing arbitrary shell injection when command strings originate from untrusted sources (e.g., ModuleConfig distributed by Manager).

**Affected Code:**
`inlong-agent/agent-common/.../utils/ExcuteLinux.java:34`

### Proposed Fix
Replace `/bin/sh -c` with `ProcessBuilder` + explicit argument arrays to eliminate shell injection vectors.

### Acceptance Criteria
- [ ] `exeCmd()` uses `ProcessBuilder` with argument arrays instead of shell string concatenation
- [ ] Existing callers work without functional regression
- [ ] Unit tests cover edge cases (special characters, spaces, pipes, etc.)

### InLong Component

InLong Agent

### Are you willing to submit PR?

- [x] Yes, I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by spiritxishi**
Fixes #12148

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

File: inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/utils/AgentUtils.java [hunk 1]
```diff
@@ -329,12 +329,11 @@ public static String fetchLocalUuid() {
                 uuid = localUuid;
                 return uuid;
             }
-            String result = ExcuteLinux.exeCmd("dmidecode | grep UUID");
-            if (StringUtils.isNotEmpty(result)
-                    && StringUtils.containsIgnoreCase(result, "UUID")) {
-                uuid = result.split(":")[1].trim();
+            String result = ExcuteLinux.exeCmd(new String[]{"dmidecode", "-s", "system-uuid"});
```
Comment (luchunliang) [inline inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/utils/AgentUtils.java]: Support "dmidecode -s " from Linux 2.10. Suggest to add a fallback.
String uuid = null;
try {
    String result = ExcuteLinux.exeCmd(new String[]{"dmidecode", "-s", "system-uuid"});
    if (StringUtils.isNotEmpty(result)) {
        result = result.trim();
        // check format
        if (result.matches("^[0-9A-Fa-f]{8}-[0-9A-Fa-f]{4}-[0-9A-Fa-f]{4}-[0-9A-Fa-f]{4}-[0-9A-Fa-f]{12}$")) {
            return result;
        }
    }
} catch (Exception e) {
    LOG.warn("dmidecode -s system-uuid failed, fallback to grep", e);
}

// Fallback
try {
    String result = ExcuteLinux.exeCmd("dmidecode | grep UUID");
    if (StringUtils.isNotEmpty(result) && StringUtils.containsIgnoreCase(result, "UUID")) {
        String[] parts = result.split(":");
        if (parts.length >= 2) {
            return parts[1].trim();
        }
    }
} catch (Exception e) {
    LOG.warn("dmidecode | grep UUID failed", e);
}

return uuid;

File: inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/utils/ExecuteLinux.java [hunk 2]
```diff
@@ -17,40 +17,377 @@
...
+ * interpreted.</p>
+ *
+ * <p>{@link #exeCmd(String)} is a legacy string-based fallback that runs commands through
+ * {@code /bin/sh -c}; it is {@link Deprecated} and must only be used with strings already
+ * validated by {@code ModuleCommandValidator}.</p>
+ *
+ * <p>Piping is emulated by chaining several {@link ProcessBuilder} instances via background
+ * threads that copy stdout to the next process's stdin, so no shell is required.</p>
  */
 public class ExcuteLinux {
```
Comment (luchunliang) [inline inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/utils/ExecuteLinux.java:55]: Please check the spelling error.

File: inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/validator/AllowedRootsResolver.java [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+     *
+     * @param p a path; it is made absolute and normalized internally before the comparison.
+     */
+    public boolean isUnderAllowedRoot(Path p) {
+        if (p == null) {
+            return false;
+        }
+        Path normalized = p.toAbsolutePath().normalize();
+        for (Path root : roots) {
+            if (normalized.startsWith(root)) {
```
Comment (luchunliang) [inline inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/validator/AllowedRootsResolver.java:239]: For symlink path, it is risk.
It is better to use toRealPath.

File: inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/validator/ModuleCommandValidator.java [hunk 4]
```diff
@@ -0,0 +1,449 @@
...
+            "rm", "cp", "mv", "mkdir", "ln", "chmod", "chown", "tar", "unzip");
+
+    private static Set<String> buildImmutableSet(String... items) {
+        Set<String> s = new HashSet<>(items.length * 2);
+        Collections.addAll(s, items);
+        return Collections.unmodifiableSet(s);
+    }
+
+    /** Substring blacklist for the metacharacter check. */
+    private static final String[] META_CHAR_BLACKLIST = new String[]{
```
Comment (luchunliang) [inline inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/validator/ModuleCommandValidator.java]: Miss these character.
private static final String[] META_CHAR_BLACKLIST = new String[]{
    "`", "$(", "${", "$",
    "&&", "||", ">>", ">", "<",
    "\\",                    
    "\u0000"                
};

File: inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/validator/ModuleCommandValidator.java [hunk 5]
```diff
@@ -0,0 +1,449 @@
...
+        }
+        return ValidationResult.okPending();
+    }
+
+    private ValidationResult validateArguments(String[] argv, String rawSegment) {
+        String cmd = argv[0];
+
+        if ("sh".equals(cmd) || "bash".equals(cmd)) {
+            for (int i = 1; i < argv.length; i++) {
+                if ("-c".equals(argv[i]) || argv[i].startsWith("-c")) {
```
Comment (luchunliang) [inline inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/validator/ModuleCommandValidator.java]: Duplication.
if (argv[i].startsWith("-c"))

File: inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/utils/ExecuteLinux.java (hunk 2 above)
Comment (spiritxishi) [reply inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/utils/ExecuteLinux.java:55]: done

File: inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/utils/AgentUtils.java (hunk 1 above)
Comment (spiritxishi) [reply inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/utils/AgentUtils.java]: done

File: inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/validator/AllowedRootsResolver.java (hunk 3 above)
Comment (spiritxishi) [reply inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/validator/AllowedRootsResolver.java:239]: done

File: inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/validator/ModuleCommandValidator.java (hunk 5 above)
Comment (spiritxishi) [reply inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/validator/ModuleCommandValidator.java]: OK remove equals

File: inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/validator/ModuleCommandValidator.java (hunk 4 above)
Comment (spiritxishi) [reply inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/validator/ModuleCommandValidator.java]: done

Comment (spiritxishi) [reply inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/validator/ModuleCommandValidator.java]: "$" appears in legitimate contexts such as "awk '{print $2}'" (field references), so it was omitted

**Review by luchunliang — DISMISSED** [review]

**Review by luchunliang — DISMISSED** [review]

**Review by vernedeng — DISMISSED** [review]

**Review by vernedeng — APPROVED** [review]

**Review by luchunliang — APPROVED** [review]
